src/controllers/moderators/auth.controller.ts

```diff
@@ -0,0 +1,61 @@
+import { Body, Controller, Get, HttpStatus, Post, Res } from '@nestjs/common';
+
+import { Response } from 'express';
+import { loginProfile, registerModerator } from 'src/dto/auth.dto';
+
+import { ModeratorAuthService } from 'src/services/moderators/auth.service';
+
+import * as jwt from 'jsonwebtoken';
+
+// moderator auth
+@Controller('moderators/auth')
+export class ModeratorAuthController {
+  constructor(private readonly authService: ModeratorAuthService) {}
+
+  // register moderator
+  @Post('register/')
+  async registerModerator(
+    @Res() response,
+    @Body() moderator: registerModerator,
+  ) {
+    try {
+      const newModerator = await this.authService.register(moderator);
```

Comment:
We don't really need try catch in controllers, the services are sufficient to handle it.